internal/buf/pool.go

```diff
@@ -188,18 +227,7 @@ func (p *Pool) Allocate(n int) Buf {
 			}
 		}
 
-		// add one more segment up to specified limit
-		if p.InitializeSegments(1) {
-			continue
-		}
-
-		// wait until some segment becomes free or some time passes, whichever comes first
-		select {
-		case <-p.segmentReleased:
-		case <-time.After(allocationRetryTimeout):
-		}
+		// add one more segment
+		p.AddSegments(1)
```

Comment:
As discussed earlier, a concern here is that there should be a way of freeing up _excess_ segments as they become empty. Perhaps keep a minimum number of segments in the pool, but releasing extra unused segments. Or perhaps having a maximum number of unused segments that can be kept around. Thoughts?

Related to this, it'd be good to track the number of unused / empty segments. That is, on `Release` increment a counter of unused segments when the allocated buffer count reaches 0, or when a segment is initially created. Decrease the counter on allocate when allocating from a currently empty segment.

Reply:
good ideas, will do in separate PR